rust_icu_uloc/src/lib.rs

```diff
@@ -63,55 +63,178 @@ impl TryFrom<&ffi::CStr> for ULoc {
+        buffered_string_char_star(versioned_function!($uloc_method), asciiz.as_ptr())
+    }};
+}
+
 impl ULoc {
     /// Implements `uloc_getLanguage`.
     pub fn language(&self) -> Result<String, common::Error> {
-        self.call_buffered_string_method(versioned_function!(uloc_getLanguage))
+        call_buffered_string_char_star_with_repr!(self, uloc_getLanguage)
     }
 
     /// Implements `uloc_getScript`.
     pub fn script(&self) -> Result<String, common::Error> {
-        self.call_buffered_string_method(versioned_function!(uloc_getScript))
+        call_buffered_string_char_star_with_repr!(self, uloc_getScript)
     }
 
     /// Implements `uloc_getCountry`.
     pub fn country(&self) -> Result<String, common::Error> {
-        self.call_buffered_string_method(versioned_function!(uloc_getCountry))
+        call_buffered_string_char_star_with_repr!(self, uloc_getCountry)
     }
 
     /// Implements `uloc_getVariant`.
     pub fn variant(&self) -> Result<String, common::Error> {
-        self.call_buffered_string_method(versioned_function!(uloc_getVariant))
+        call_buffered_string_char_star_with_repr!(self, uloc_getVariant)
     }
 
     /// Implements `uloc_canonicalize` from ICU4C.
     pub fn canonicalize(&self) -> Result<ULoc, common::Error> {
-        self.call_buffered_string_method(versioned_function!(uloc_canonicalize))
-            .map(|repr| ULoc { repr })
+        call_buffered_string_char_star_with_repr!(self, uloc_canonicalize).map(|repr| ULoc { repr })
     }
 
     /// Implements `uloc_addLikelySubtags` from ICU4C.
     pub fn add_likely_subtags(&self) -> Result<ULoc, common::Error> {
-        self.call_buffered_string_method(versioned_function!(uloc_addLikelySubtags))
+        call_buffered_string_char_star_with_repr!(self, uloc_addLikelySubtags)
             .map(|repr| ULoc { repr })
     }
 
     /// Implements `uloc_minimizeSubtags` from ICU4C.
     pub fn minimize_subtags(&self) -> Result<ULoc, common::Error> {
-        self.call_buffered_string_method(versioned_function!(uloc_minimizeSubtags))
+        call_buffered_string_char_star_with_repr!(self, uloc_minimizeSubtags)
             .map(|repr| ULoc { repr })
     }
 
+    // Implements `uloc_toLanguageTag` from ICU4C.
+    pub fn to_language_tag(&self, strict: bool) -> Result<String, common::Error> {
+        let repr = self.as_c_string()?;
+        let strict = if strict { 1 } else { 0 };
+        buffered_string_char_star_ubool(
+            versioned_function!(uloc_toLanguageTag),
+            repr.as_ptr(),
+            strict,
+        )
+    }
+
     /// Returns the current label of this locale.
     pub fn label(&self) -> &str {
         &self.repr
     }
 
     /// Returns the current locale name as a C string.
-    pub fn as_c_str(&self) -> ffi::CString {
-        ffi::CString::new(self.repr.clone()).expect("ULoc contained interior NUL bytes")
+    pub fn as_c_string(&self) -> Result<ffi::CString, common::Error> {
```

Comment:
This is a backward incompatible API change, so we can not simply kill the method as that would break all users. 
 
Prefer to add a method instead, and mark this one as deprecated.

Reply:
Reverted.